EPOParser.py

```diff
@@ -4,115 +4,133 @@
+    return lambda el: getattr(el, info)
 
 # tree :: String -> Tree
 def tree(filename):
-  return etree.parse(filename, etree.XMLParser(encoding='utf-8', recover=True))
-
-# getFileInfo :: String -> [(filename, file-location)]
-def getFileInfo(indexFilename):
-  return fzip(
-    fmap(fromElement('text'), getElement(tree(indexFilename), '//filename')),
-    fmap(fromElement('text'), getElement(tree(indexFilename), '//file-location'))
-  )
-
-# getElement :: Tree, xpath -> [Element]
-def getElement(tree, elementPath):
-  return tree.xpath(elementPath)
-
-# getMetadata :: Tree, xpath, attrib | text, String -> [] | [(alias, tag, {} | String)]
-def getMetadata(tree, elementPath, info, alias):
-  return fmap(lambda el: ( alias, el.tag, getattr(el, info) ), getElement(tree, elementPath))
+    """
+    This is a multi-line docstring.
+
+    It is used when there is more info that should be shared than fits on one
+    line.
+    """
+    return etree.parse(
+        filename, etree.XMLParser(encoding='utf-8', recover=True))
+
+# get_file_info :: String -> [(filename, file-location)]
+def get_file_info(indexFilename):
+    return fzip(
+        fmap(from_element('text'), get_element(tree(indexFilename), '//filename')),
+        fmap(from_element('text'), get_element(tree(indexFilename), '//file-location'))
+    )
+
+# get_element :: Tree, xpath -> [Element]
+def get_element(tree, elementPath):
+    return tree.xpath(elementPath)
+
+# get_metadata :: Tree, xpath, attrib | text, String -> [] | [(alias, tag, {} | String)]
+def get_metadata(tree, elementPath, info, alias):
+    return fmap(lambda el: ( alias, el.tag, getattr(el, info) ), get_element(tree, elementPath))
 
 # parse :: Tree, [[ xpath, attrib | text, String ]] -> [(alias, tag {} | String)]
 def parse(tree, fields):
-  return fcompose(
-    fcmap(lambda x: x[0]),
-    fcfilter(lambda x: bool(x)),
-    fcmap(lambda field: getMetadata(tree, field[0], field[1], field[2]))
-  )(fields)
+    return fcompose(
+        fcmap(lambda x: x[0]),
+        fcfilter(lambda x: bool(x)),
+        fcmap(lambda field: get_metadata(tree, field[0], field[1], field[2]))
+    )(fields)
 
 ##############
 # processing #
 ##############
 
 def process(fields, state, callback, xmlFile, archiveFile):
-  if not os.path.isfile(marker(xmlFile)):
-    zipfile.ZipFile(archiveFile, "r").extract(xmlFile)
-    
+    if not os.path.isfile(marker(xmlFile)):
+        zipfile.ZipFile(archiveFile, "r").extract(xmlFile)
+
     # add validation/testing
-    
-    metadata = parse( tree(xmlFile), fields() )
-    
+
+    metadata = parse(tree(xmlFile), fields())
+
     # send off to queue (handle failures and restarts), if success, write file...
     callback(metadata)
-    
+
     # gives us a durable state if the script breaks and reruns
-    completefile = open(marker(xmlFile),"w") 
+    completefile = open(marker(xmlFile), "w")
     completefile.write(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
```

Comment:
This is an example of a library that you could have imported less verbosely. Either of these are options:

  * `import datetime as dt`. Then you would call the less verbose `dt.datetime.now()`
  * `from datetime import datetime`. Since you only use the `datetime` module's `datetime` submodule (unfortunate naming on their part) you would now be able to call this as simple `datetime.now()`. Long-term this might be risky if you were to need other submodules of the top-level `datetime` module.